examples/wallet/app/actions/router.js

```diff
@@ -3,10 +3,17 @@ import { push } from 'connected-react-router';
 import type { Dispatch, GetState } from '../reducers/types';
 import type { Address } from '../models';
 import routes from '../constants/routes';
+import { setAccountFromPrivateKey } from './account';
+
+import { readAccountKeysFromLocalStorage } from '../utils/storage';
 
 // eslint-disable-next-line import/prefer-default-export
 export const redirectToFirstAppPage = () => {
   return (dispatch: Dispatch, getState: GetState) => {
+    const accountKeys = readAccountKeysFromLocalStorage();
+    if (accountKeys !== undefined) {
+      return dispatch(setAccountFromPrivateKey(accountKeys.privateKey));
```

Comment:
this generates the whole AccountKeys object (which you already have), extracts the private key, and then uses it to call an action which uses the wasm bindings to generate the AccountKeys object from the private key.

Dispatch the `SET_KEYS` action directly instead, perhaps via a function as is shown in the redux tutorials.

Reply:
Corrected and a new function to update account state and transactions list is added